go/host/db/batches.go

```diff
@@ -32,20 +32,23 @@ func (db *DB) GetBatchHeader(hash gethcommon.Hash) (*common.Header, error) {
 }
 
 // AddBatchHeader adds a batch's header to the known headers
-func (db *DB) AddBatchHeader(header *common.Header, txHashes []common.TxHash) error {
+func (db *DB) AddBatchHeader(batch *common.ExtBatch) error {
 	b := db.kvStore.NewBatch()
 
-	if err := db.writeBatchHeader(header); err != nil {
+	if err := db.writeBatchHeader(batch.Header); err != nil {
 		return fmt.Errorf("could not write batch header. Cause: %w", err)
 	}
-	if err := db.writeBatchTxHashes(b, header.Hash(), txHashes); err != nil {
+	if err := db.writeBatch(batch); err != nil {
```

Comment:
To avoid the round trip to the enclave, I store the full `common.ExtBatch` locally on the host.